how-to/customize-workspace/client/src/framework/platform/interopbroker.ts

```diff
@@ -395,6 +412,42 @@ export function interopOverride(
 				throw new Error(RESOLVER_TIMEOUT);
 			}
 		}
+
+		public async fdc3HandleOpen(
+			fdc3OpenOptions: { app: PlatformApp | string; context: OpenFin.Context },
+			clientIdentity: OpenFin.ClientIdentity
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		): Promise<any> {
+			if (fdc3OpenOptions?.app === undefined || fdc3OpenOptions?.app === null) {
+				logger.error("A request to fdc3.open did not pass an fdc3OpenOptions object");
+				throw new Error(OPEN_APP_NOT_FOUND);
+			}
+
+			const requestedId =
+				typeof fdc3OpenOptions.app === "string"
+					? fdc3OpenOptions.app
+					: fdc3OpenOptions.app.appId || fdc3OpenOptions.app.name;
```

Comment:
`??`

Reply:
best to discuss over voice, but the fdc3 open api can be passed a name. An alternative could be to extend App just in the open signature and use appId or name against our App.appid (if appId is empty in the mappers as fdc3 has name as the mandatory property for an app).

Reply:
Me being lazy, I mean use the `??` operator :)